File: shinebuttonlib/src/main/java/com/sackcentury/shinebuttonlib/EasingInterpolator.java
package com.sackcentury.shinebuttonlib;

import android.animation.TimeInterpolator;

import androidx.annotation.NonNull;

/**
 * @author xiongke
 * @date 2019-06-11
 */
public class EasingInterpolator implements TimeInterpolator {

    private final Ease ease;

    public EasingInterpolator(@NonNull Ease ease) {
        this.ease = ease;
    }

    @Override
    public float getInterpolation(float input) {
        return EasingProvider.get(this.ease, input);
    }

    public Ease getEase() {
        return ease;
    }
}

File: shinebuttonlib/src/main/java/com/sackcentury/EasingInterpolator.kt
package com.sackcentury

import android.animation.TimeInterpolator
import com.sackcentury.Ease

/**
 * @author xiongke
 * @since 2026/1/14
 **/
class EasingInterpolator(private val ease: Ease) : TimeInterpolator {

    override fun getInterpolation(input: Float): Float {
        return EasingProvider.get(ease, input)
    }
}
